packages/sdk-core/src/BacktraceCoreClient.ts

```diff
@@ -117,31 +115,31 @@ export abstract class BacktraceCoreClient {
         ]);
         this.attachments = options.attachments ?? [];
 
-        if (databaseStorageProvider && options?.database?.enabled === true) {
-            this._database = new BacktraceDatabase(
-                this.options.database,
-                databaseStorageProvider,
-                this._reportSubmission,
-            );
-            this._database.start();
+        if (options?.database?.enabled === true) {
+            this._database = new BacktraceDatabase(this.options.database, this._reportSubmission);
         }
 
         this._rateLimitWatcher = new RateLimitWatcher(options.rateLimit);
 
         const metrics = new MetricsBuilder(options, _sessionProvider, this._attributeProvider, requestHandler).build();
         if (metrics) {
             this._metrics = metrics;
-            this._metrics.start();
         }
 
         if (options.breadcrumbs?.enable !== false) {
             this.breadcrumbsManager = new BreadcrumbsManager(options?.breadcrumbs, breadcrumbsSetup);
             this._attributeProvider.addProvider(this.breadcrumbsManager);
             this.attachments.push(this.breadcrumbsManager.breadcrumbsStorage);
-            this.breadcrumbsManager.start();
         }
     }
 
+    public initialize() {
```

Comment:
shouldn't we move more code from the constructor to the initialize method? For example metrics startup?